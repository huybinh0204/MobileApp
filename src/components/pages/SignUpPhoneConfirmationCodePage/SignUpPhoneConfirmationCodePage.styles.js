import styled from '@emotion/native';
import { BaseText } from '_components/atoms';

export const Container = styled.View(({ theme }) => ({
  flex: 1,
  padding: theme.spacing.m,
  justifyContent: 'space-between',
}));

export const ConfirmPhoneTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.l,
  textAlign: 'left',
}));
