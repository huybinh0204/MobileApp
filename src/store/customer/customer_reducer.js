import { createActions, createReducer } from 'reduxsauce';
import Immutable from 'seamless-immutable';

const { Types, Creators } = createActions(
  {
    setExternalId: ['externalId'],
    updateCustomerData: ['newData'],
    fetchCustomer: ['customerExternalId'],
    setFetchCustomerSuccess: ['data'],
    setFetchCustomerError: ['fetchCustomerError'],
    updateAddress: ['customerExternalId', 'address'],
    updateEmail: ['customerExternalId', 'email'],
    setUpdateAddressSuccess: ['updateAddressSuccess'],
    setUpdateAddressError: ['updateAddressError'],
    setUpdateEmailSuccess: ['updateEmailSuccess'],
    setUpdateEmailError: ['updateEmailError'],
    startPhoneVerification: ['customerExternalId', 'phoneNumber'],
    setPhoneVerificationSuccess: ['phoneVerificationSuccess'],
    setPhoneVerificationError: ['phoneVerificationError'],
    confirmPhoneVerification: ['code', 'customerExternalId', 'phoneNumber'],
    setConfirmPhoneVerificationSuccess: ['confirmPhoneVerificationSuccess'],
    setConfirmPhoneVerificationError: ['confirmPhoneVerificationError'],
    setIsLoading: ['isLoading'],
    checkVerificationStatus: ['field', 'customerExternalId'],
    setVerificationStatus: ['verificationStatus'],
    setVerificationStatusError: ['verificationError'],
    getSuggestedAddresses: ['address'],
    setSuggestedAddresses: ['suggestedAddresses'],
    setSuggestedAddressesError: ['suggestedAddressesError'],
    resetCustomer: [null],
  },
  {
    prefix: '@@BE-CUSTOMER/',
  }
);

export const CustomerActions = Creators;
export const CustomerTypes = Types;

// INITIAL STATE
export const INITIAL_STATE = Immutable({
  externalId: null,
  data: null,
  fetchCustomerError: null,
  updateAddressError: null,
  updateAddressSuccess: false,
  updateEmailError: null,
  updateEmailSuccess: false,
  phoneVerificationSuccess: false,
  phoneVerificationError: null,
  confirmPhoneVerificationSuccess: false,
  confirmPhoneVerificationError: null,
  isLoading: false,
  verificationStatus: null,
  verificationError: null,
  suggestedAddresses: [],
  suggestedAddressesError: null,
});

// SELECTORS
export const CustomerSelectors = {
  getExternalId: (state) => state.customer.externalId,
  getCustomer: (state) => state.customer.data,
  getFetchCustomerError: (state) => state.customer.fetchCustomerError,
  getUpdateAddressError: (state) => state.customer.updateAddressError,
  getUpdateAddressSuccess: (state) => state.customer.updateAddressSuccess,
  getUpdateEmailError: (state) => state.customer.updateEmailError,
  getUpdateEmailSuccess: (state) => state.customer.updateEmailSuccess,
  getPhoneVerificationSuccess: (state) => state.customer.phoneVerificationSuccess,
  getPhoneVerificationError: (state) => state.customer.phoneVerificationError,
  getConfirmPhoneVerificationSuccess: (state) => state.customer.confirmPhoneVerificationSuccess,
  getConfirmPhoneVerificationError: (state) => state.customer.confirmPhoneVerificationError,
  getIsLoading: (state) => state.customer.isLoading,
  getIsRefreshingCustomer: (state) => state.customer.isLoading && state.customer.data !== null,
  getVerificationStatus: (state) => state.customer.verificationStatus,
  getVerificationStatusError: (state) => state.customer.verificationError,
  getSuggestedAddresses: (state) => state.customer.suggestedAddresses,
  getSuggestedAddressesError: (state) => state.customer.suggestedAddressesError,
};

// ACTION HANDLERS

const updateCustomerData = (state, { newData }) => {
  return state.update('data', (data) => ({ ...data, ...newData }));
};

const setExternalId = (state, { externalId }) => {
  return state.merge({ externalId });
};

const setFetchCustomerSuccess = (state, { data }) => {
  return state.merge({ data });
};

const setFetchCustomerError = (state, { fetchCustomerError }) => {
  return state.merge({ fetchCustomerError });
};

const setUpdateAddressSuccess = (state, { updateAddressSuccess }) => {
  return state.merge({ updateAddressSuccess });
};

const setUpdateAddressError = (state, { updateAddressError }) => {
  return state.merge({ updateAddressError });
};

const setIsLoading = (state, { isLoading }) => {
  return state.merge({ isLoading });
};

const setUpdateEmailSuccess = (state, { updateEmailSuccess }) => {
  return state.merge({ updateEmailSuccess });
};

const setUpdateEmailError = (state, { updateEmailError }) => {
  return state.merge({ updateEmailError });
};

const setPhoneVerificationSuccess = (state, { phoneVerificationSuccess }) => {
  return state.merge({ phoneVerificationSuccess });
};

const setPhoneVerificationError = (state, { phoneVerificationError }) => {
  return state.merge({ phoneVerificationError });
};

const setConfirmPhoneVerificationSuccess = (state, { confirmPhoneVerificationSuccess }) => {
  return state.merge({ confirmPhoneVerificationSuccess });
};

const setConfirmPhoneVerificationError = (state, { confirmPhoneVerificationError }) => {
  return state.merge({ confirmPhoneVerificationError });
};

const setVerificationStatus = (state, { verificationStatus }) => {
  return state.merge({ verificationStatus });
};
const setVerificationStatusError = (state, { verificationError }) => {
  return state.merge({ verificationError });
};

const setSuggestedAddresses = (state, { suggestedAddresses }) => {
  return state.merge({
    suggestedAddresses: suggestedAddresses.map((address) => ({
      addressLine1: address.primary_line,
      city: address.city,
      state: address.state,
      zipCode: address.zip_code,
    })),
  });
};

const setSuggestedAddressesError = (state, { suggestedAddressesError }) => {
  return state.merge({ suggestedAddressesError });
};

const resetCustomer = () => {
  return INITIAL_STATE;
};

// REDUCER
export const CustomerReducer = createReducer(INITIAL_STATE, {
  [Types.UPDATE_CUSTOMER_DATA]: updateCustomerData,
  [Types.SET_EXTERNAL_ID]: setExternalId,
  [Types.SET_FETCH_CUSTOMER_SUCCESS]: setFetchCustomerSuccess,
  [Types.SET_FETCH_CUSTOMER_ERROR]: setFetchCustomerError,
  [Types.SET_UPDATE_ADDRESS_SUCCESS]: setUpdateAddressSuccess,
  [Types.SET_UPDATE_ADDRESS_ERROR]: setUpdateAddressError,
  [Types.SET_UPDATE_EMAIL_SUCCESS]: setUpdateEmailSuccess,
  [Types.SET_UPDATE_EMAIL_ERROR]: setUpdateEmailError,
  [Types.SET_IS_LOADING]: setIsLoading,
  [Types.RESET_CUSTOMER]: resetCustomer,
  [Types.SET_PHONE_VERIFICATION_SUCCESS]: setPhoneVerificationSuccess,
  [Types.SET_PHONE_VERIFICATION_ERROR]: setPhoneVerificationError,
  [Types.SET_CONFIRM_PHONE_VERIFICATION_SUCCESS]: setConfirmPhoneVerificationSuccess,
  [Types.SET_CONFIRM_PHONE_VERIFICATION_ERROR]: setConfirmPhoneVerificationError,
  [Types.SET_VERIFICATION_STATUS]: setVerificationStatus,
  [Types.SET_VERIFICATION_STATUS_ERROR]: setVerificationStatusError,
  [Types.SET_SUGGESTED_ADDRESSES]: setSuggestedAddresses,
  [Types.SET_SUGGESTED_ADDRESSES_ERROR]: setSuggestedAddressesError,
});
