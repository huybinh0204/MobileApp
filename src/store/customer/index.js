export * from './customer_reducer';
export * from './customer_sagas';
