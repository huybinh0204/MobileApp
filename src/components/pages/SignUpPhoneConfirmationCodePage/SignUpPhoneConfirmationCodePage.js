import { useNavigation, useRoute } from '@react-navigation/native';
import React, { useEffect, useRef, useState } from 'react';
import { View } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { MainButton, SegmentedTextInput, Toast } from '_components/atoms';
import { OnboardingScreenLayout } from '_components/organisms';
import { NAVIGATION, PHONE_CODES, TOAST_TYPES } from '_constants';
import strings from '_localization/index';
import { CustomerActions, CustomerSelectors } from '_store/customer';
import { cleanPhoneNumber, formatPhoneNumber } from '_utilities/Account';
import { ConfirmPhoneTitle, Container } from './SignUpPhoneConfirmationCodePage.styles';

const CELL_COUNT = 6;
const RESEND_TIMEOUT = 15000; // 15 seconds

const SignUpPhoneConfirmationCodePage = () => {
  const { params } = useRoute();
  const { navigate } = useNavigation();
  const dispatch = useDispatch();

  const customerExternalId = useSelector(CustomerSelectors.getExternalId);
  const confirmSuccess = useSelector(CustomerSelectors.getConfirmPhoneVerificationSuccess);
  const confirmError = useSelector(CustomerSelectors.getConfirmPhoneVerificationError);
  const isLoading = useSelector(CustomerSelectors.getIsLoading);

  const timer = useRef(null);
  const [code, setCode] = useState('');
  const [showResendButton, setShowResendButton] = useState(false);

  const localizedPhoneNumber = cleanPhoneNumber(PHONE_CODES.US, params?.phoneNumber);

  useEffect(() => {
    if (confirmSuccess) {
      dispatch(CustomerActions.setConfirmPhoneVerificationSuccess(false));
      dispatch(CustomerActions.setConfirmPhoneVerificationError(null));
      navigate(NAVIGATION.auth.signUpName, { ...params });
    }
  }, [confirmSuccess, navigate, params, dispatch]);

  useEffect(() => {
    if (code?.length < CELL_COUNT) {
      return;
    }

    dispatch(
      CustomerActions.confirmPhoneVerification(code, customerExternalId, localizedPhoneNumber)
    );

    if (timer?.current) {
      clearTimeout(timer.current);
    }
  }, [code, dispatch, localizedPhoneNumber, customerExternalId]);

  const resetTimer = () => {
    if (timer?.current) {
      clearTimeout(timer.current);
    }
    timer.current = setTimeout(() => {
      setShowResendButton(true);
    }, RESEND_TIMEOUT);
  };

  const resendCode = () => {
    dispatch(CustomerActions.startPhoneVerification(customerExternalId, localizedPhoneNumber));
    setShowResendButton(false);
    setCode('');
    resetTimer();
  };

  const handleCloseToast = () => {
    dispatch(CustomerActions.setConfirmPhoneVerificationSuccess(false));
    dispatch(CustomerActions.setConfirmPhoneVerificationError(null));
  };

  return (
    <OnboardingScreenLayout step={5} testID="SignUpPhonePage">
      <Container>
        <View>
          <ConfirmPhoneTitle>
            {strings.formatString(strings.signUp.confirmPhoneNumber.title, {
              phone: formatPhoneNumber(params?.phoneNumber),
            })}
          </ConfirmPhoneTitle>
          <SegmentedTextInput
            autoFocus
            cellCount={CELL_COUNT}
            keyboardType="phone-pad"
            setValue={setCode}
            value={code}
            testID="SMSCode"
          />
        </View>
        <MainButton
          disabled={!showResendButton}
          onPress={resendCode}
          isLoading={isLoading}
          testID="signUpPhoneSubmitButton"
          variant="link"
        >
          {strings.signUp.confirmPhoneNumber.resendCode}
        </MainButton>
      </Container>
      <Toast
        type={TOAST_TYPES.ERROR}
        header={confirmError?.title}
        content={confirmError?.description}
        onClose={handleCloseToast}
        show={confirmError !== null}
        testID="confirmCodeErrorToast"
      />
    </OnboardingScreenLayout>
  );
};

export default SignUpPhoneConfirmationCodePage;
