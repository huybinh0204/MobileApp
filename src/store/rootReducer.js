import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
