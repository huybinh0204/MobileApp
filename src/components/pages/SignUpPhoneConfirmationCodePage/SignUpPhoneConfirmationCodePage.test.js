import { useNavigation } from '@react-navigation/native';
import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { useDispatch } from 'react-redux';
import { createStore } from 'redux';
import { CustomerActions } from '_store/customer';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import SignUpPhoneConfirmationCodePage from './SignUpPhoneConfirmationCodePage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('SignUpPhoneConfirmationCodePage', () => {
  let mockStore;
  let mockNavigation;
  let mockDispatch;
  let params;

  beforeEach(() => {
    mockDispatch = jest.fn();
    mockNavigation = navigationMock(jest.fn);
    mockStore = createStore(rootReducer, {
      customer: {
        externalId: 'mocked-external-id',
      },
    });
    params = {
      phoneNumber: '5104831826',
    };
    useNavigation.mockImplementation(() => mockNavigation);
    useDispatch.mockReturnValue(mockDispatch);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should trigger the change of screen if the code has 6 digits', async () => {
    mockStore = createStore(rootReducer, {
      customer: {
        externalId: 'mocked-external-id',
        confirmPhoneVerificationSuccess: true,
      },
    });

    const { getByTestId } = render(
      <ComponentWithProviders
        component={SignUpPhoneConfirmationCodePage}
        store={mockStore}
        params={params}
      />
    );

    const codeInput = getByTestId('SMSCode');
    fireEvent.changeText(codeInput, '510483');

    expect(mockDispatch).toBeCalledWith(
      CustomerActions.confirmPhoneVerification('510483', 'mocked-external-id', '+15104831826')
    );
  });

  it('should not navigate if the code input does not have 6 digits', async () => {
    render(
      <ComponentWithProviders
        component={SignUpPhoneConfirmationCodePage}
        store={mockStore}
        params={params}
      />
    );

    expect(mockDispatch).toBeCalledTimes(0);
  });

  it('should match snapshot', () => {
    const { getByTestId, toJSON } = render(
      <ComponentWithProviders
        component={SignUpPhoneConfirmationCodePage}
        store={mockStore}
        params={params}
      />
    );

    const codeInput = getByTestId('SMSCode');

    fireEvent.changeText(codeInput, '510483');

    expect(toJSON()).toMatchSnapshot();
  });
});
