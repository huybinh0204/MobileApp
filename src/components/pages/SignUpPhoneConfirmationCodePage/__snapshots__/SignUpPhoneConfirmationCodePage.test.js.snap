// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPhoneConfirmationCodePage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="SignUpPhonePage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "width": "50%",
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "center",
                                    "backgroundColor": "#F2F3F4",
                                    "borderColor": "#000000",
                                    "borderRadius": 10,
                                    "height": 9,
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "backgroundColor": "#3D3397",
                                    "borderRadius": 10,
                                    "bottom": 0,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                    "width": "0%",
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "justifyContent": "space-between",
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <View>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 36,
                                      "lineHeight": 50.5,
                                      "marginBottom": 87,
                                      "textAlign": "left",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Please enter the code sent to
(510) 483-1826
                              </Text>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                  }
                                }
                              >
                                <View
                                  onLayout={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#D4D7DB",
                                        "borderRadius": 6,
                                        "borderWidth": 2,
                                        "zIndex": 0,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#1F1D25",
                                          "fontFamily": "Poppins",
                                          "fontSize": 36,
                                          "height": 136,
                                          "lineHeight": 87,
                                          "paddingTop": 29,
                                          "textAlign": "center",
                                          "width": 87,
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    5
                                  </Text>
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#D4D7DB",
                                        "borderRadius": 6,
                                        "borderWidth": 2,
                                        "zIndex": 0,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#1F1D25",
                                          "fontFamily": "Poppins",
                                          "fontSize": 36,
                                          "height": 136,
                                          "lineHeight": 87,
                                          "paddingTop": 29,
                                          "textAlign": "center",
                                          "width": 87,
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    1
                                  </Text>
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#D4D7DB",
                                        "borderRadius": 6,
                                        "borderWidth": 2,
                                        "zIndex": 0,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#1F1D25",
                                          "fontFamily": "Poppins",
                                          "fontSize": 36,
                                          "height": 136,
                                          "lineHeight": 87,
                                          "paddingTop": 29,
                                          "textAlign": "center",
                                          "width": 87,
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    0
                                  </Text>
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#D4D7DB",
                                        "borderRadius": 6,
                                        "borderWidth": 2,
                                        "zIndex": 0,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#1F1D25",
                                          "fontFamily": "Poppins",
                                          "fontSize": 36,
                                          "height": 136,
                                          "lineHeight": 87,
                                          "paddingTop": 29,
                                          "textAlign": "center",
                                          "width": 87,
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    4
                                  </Text>
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#D4D7DB",
                                        "borderRadius": 6,
                                        "borderWidth": 2,
                                        "zIndex": 0,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#1F1D25",
                                          "fontFamily": "Poppins",
                                          "fontSize": 36,
                                          "height": 136,
                                          "lineHeight": 87,
                                          "paddingTop": 29,
                                          "textAlign": "center",
                                          "width": 87,
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    8
                                  </Text>
                                </View>
                                <View
                                  onLayout={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#D4D7DB",
                                        "borderRadius": 6,
                                        "borderWidth": 2,
                                        "zIndex": 0,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#1F1D25",
                                          "fontFamily": "Poppins",
                                          "fontSize": 36,
                                          "height": 136,
                                          "lineHeight": 87,
                                          "paddingTop": 29,
                                          "textAlign": "center",
                                          "width": 87,
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    3
                                  </Text>
                                </View>
                                <TextInput
                                  autoCapitalize="characters"
                                  autoCorrect={false}
                                  autoFocus={true}
                                  blurOnSubmit={false}
                                  caretHidden={true}
                                  clearButtonMode="never"
                                  disableFullscreenUI={true}
                                  keyboardType="phone-pad"
                                  maxLength={6}
                                  onBlur={[Function]}
                                  onChangeText={[Function]}
                                  onFocus={[Function]}
                                  onPressOut={[Function]}
                                  spellCheck={false}
                                  style={
                                    Object {
                                      "bottom": 0,
                                      "fontSize": 1,
                                      "left": 0,
                                      "opacity": 0.01,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    }
                                  }
                                  testID="SMSCode"
                                  underlineColorAndroid="transparent"
                                  value="510483"
                                />
                              </View>
                            </View>
                            <View
                              accessibilityState={
                                Object {
                                  "disabled": true,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignSelf": "stretch",
                                  "backgroundColor": "transparent",
                                  "borderRadius": 10,
                                  "opacity": 1,
                                  "paddingHorizontal": 45.5,
                                  "paddingVertical": 31,
                                }
                              }
                              testID="signUpPhoneSubmitButton"
                            >
                              <Text
                                allowFontScaling={false}
                                disabled={true}
                                style={
                                  Array [
                                    Object {
                                      "color": "#D4D7DB",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Resend Code
                              </Text>
                            </View>
                          </View>
                          <View
                            onLayout={[Function]}
                            paddingHorizontal={null}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "bottom": 0,
                                  "paddingHorizontal": 58,
                                  "position": "absolute",
                                  "width": "100%",
                                },
                                Object {
                                  "opacity": 1,
                                  "transform": Array [
                                    Object {
                                      "translateY": AnimatedValue {
                                        " __value": 0,
                                      },
                                    },
                                  ],
                                },
                              ]
                            }
                            testID="confirmCodeErrorToast"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "#FFFFFF",
                                    "borderColor": "#F2F3F4",
                                    "borderRadius": 10,
                                    "borderWidth": 0,
                                    "padding": 29,
                                    "shadowColor": "#362E60",
                                    "shadowOffset": Object {
                                      "height": 7,
                                      "width": 0,
                                    },
                                    "shadowOpacity": 0.1,
                                    "shadowRadius": 10,
                                  },
                                  Array [
                                    Object {
                                      "backgroundColor": "#D02B35",
                                      "borderColor": "#D02B35",
                                      "borderWidth": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ],
                                ]
                              }
                              type="error"
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "flex-start",
                                      "flexDirection": "row",
                                      "justifyContent": "space-between",
                                    },
                                    undefined,
                                  ]
                                }
                                testID="toastHeader"
                              >
                                <SvgMock
                                  height={36}
                                  width={36}
                                />
                                <Text
                                  allowFontScaling={false}
                                  numberOfLines={2}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 21.5,
                                        "lineHeight": 30.5,
                                        "paddingHorizontal": 18,
                                        "paddingTop": 2,
                                      },
                                      undefined,
                                    ]
                                  }
                                  type="error"
                                />
                                <View
                                  accessibilityLabel="Close toast"
                                  accessibilityRole="button"
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                  testID="toastCloseButton"
                                >
                                  <SvgMock
                                    height={20}
                                    style={
                                      Array [
                                        Object {
                                          "height": 65,
                                          "width": 65,
                                        },
                                        undefined,
                                      ]
                                    }
                                    type="error"
                                    width={20}
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
